### PR TITLE
fix(agent): don't add blocks that contain only punctuation

### PR DESCRIPTION
**Pull request by douglas-reid**
Trailing punctuation blocks are of little to no value (imho). This PR ensures that we don't add those blocks.

From:

```
Here is the image you requested: 
(image/png: B1A98152-A422-4B44-8FD9-D7C954E82D2A)
.
```

To:

```
Here is the image you requested: 
(image/png: B1A98152-A422-4B44-8FD9-D7C954E82D2A)
```

### CODE REVIEW

File: src/steamship/agents/functional/output_parser.py [hunk 1]
```diff
@@ -1,13 +1,17 @@
...
 from typing import Dict, List, Optional
 
 from steamship import Block, MimeTypes, Steamship
 from steamship.agents.schema import Action, AgentContext, FinishAction, OutputParser, Tool
 from steamship.data.tags.tag_constants import RoleTag
 
 
-# TODO(dougreid): extract shared bits from this and the ReACT output parser into a utility?
+def is_punctuation(text: str):
+    return text in string.punctuation
```
Comment (dkolas) [inline src/steamship/agents/functional/output_parser.py:12]: Do you want this to return true if ALL the text is punctuation, or only if it is a single character of punctuation?  The current implementation does the latter:
```
>>> s = ".."
>>> import string
>>> s in string.punctuation
False
```

**Review by dkolas — APPROVED** [review]
Looks good. One question below.  I'm good either way, just whatever you think best.

File: src/steamship/agents/functional/output_parser.py (hunk 1 above)
Comment (douglas-reid) [reply src/steamship/agents/functional/output_parser.py]: good point. updated. still doesn't filter strings like `....   ....   ...`, but that is so degenerate a case that I'm not sure we need to worry about it.

**Comment by douglas-reid** [comment]
PTAL.

**Pull request merged**